extensions/ql-vscode/src/view/common/TextButton.tsx

```diff
@@ -8,19 +8,29 @@ const StyledButton = styled.button<{ size: Size }>`
   color: var(--vscode-textLink-foreground);
   border: none;
   cursor: pointer;
-  font-size: ${(props) => props.size};
+  font-size: ${(props) => props.size ?? "1em"};
+  padding: 0;
 `;
 
 const TextButton = ({
   size,
   onClick,
+  className,
+  title,
   children,
 }: {
-  size: Size;
+  size?: Size;
   onClick: () => void;
+  className?: string;
+  title?: string;
   children: React.ReactNode;
 }) => (
-  <StyledButton size={size} onClick={onClick}>
+  <StyledButton
+    size={size}
+    onClick={onClick}
+    className={className}
```

Comment:
Apologies for the drive-by comment, just a thought: Seems a bit of a shame to be changing our nice, self-contained button and adding this reliance to external CSS classes. Should we create a new component instead? It'd be easier to remove it later on than changing this one again I think.

Reply:
This is not actually used for passing the `className` directly to the component, but is used so we can call `styled(TextButton)` to add to the styles of this component. We should never be passing `className` directly, but being able to augment the styles is really useful: https://github.com/github/vscode-codeql/pull/2719/files#diff-919ab8acd98848b2bd69fe5ef5b98c920c6fdf5db48cf04e9e259d333c0fa6b6R17-R21

Reply:
Oh right. That's some confusing syntax!